modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBDecision.java

```diff
@@ -121,8 +180,17 @@ public Expression decisionGreaterEqualThan(){
      *          [""all"", boolean, boolean, ...]: boolean
      * @return
      */
-    public Expression decisionAll(){
-        return null;
+    private Expression decisionAll() {
+        // validate the arg list
+        if (json.size() < 2) {
+            throwInsufficientArgumentCount(ALL, 1);
+        }
+        Expression[] expressions = new Expression[json.size()-1];
+        for (int i = 1; i < json.size(); ++i) {
+            Expression expression = parse.string(json, i);
+            expressions[i-1] = expression;
+        }
+        return ff.function("mbAll", expressions);
```

Comment:
This should be equivalent to a sequence of ANDs - lets not add functions where we don't need too.

Reply:
I was not able to come up with a way to reuse the GeoTools `And` filter (org.geotools.filter.AndImpl) in this case and produce an expression that could be evaluated at a later time. I also was not able to find an existing Expression-compatible Filter function that did this, hence the custom implementation.